{
    "status": "Fix Released", 
    "last_updated": "2013-09-25 18:51:07.845752+00:00", 
    "description": "nova.compute.manager calls methods in nova.volume.manager directly.\nThis must be via api module.\n\n146:        self.volume_manager = utils.import_object(FLAGS.volume_manager)\n1230:        dev_path = self.volume_manager.setup_compute_volume(context, volume_id)\n1241:        dev_path = self.volume_manager.setup_compute_volume(context,\n1267:            self.volume_manager.remove_compute_volume(context,\n1289:        self.volume_manager.remove_compute_volume(context, volume_id)\n1306:        self.volume_manager.remove_compute_volume(context, volume_id)\n1401:                self.volume_manager.setup_compute_volume(context, v['id'])\n1511:                self.volume_manager.remove_compute_volume(ctxt, vol['id'])", 
    "tags": [
        "diablo-backport"
    ], 
    "importance": "Undecided", 
    "heat": 8, 
    "link": "https://bugs.launchpad.net/nova/+bug/872445", 
    "owner": "https://api.launchpad.net/1.0/~vishvananda", 
    "id": 872445, 
    "index": 3577, 
    "openned": "2011-10-11 19:08:49.316558+00:00", 
    "created": "2011-10-11 19:07:40.379268+00:00", 
    "title": "nova.compute.manager calls methods in nova.volume.manager directly", 
    "comments": [
        {
            "content": "nova.compute.manager calls methods in nova.volume.manager directly.\nThis must be via api module.\n\n146:        self.volume_manager = utils.import_object(FLAGS.volume_manager)\n1230:        dev_path = self.volume_manager.setup_compute_volume(context, volume_id)\n1241:        dev_path = self.volume_manager.setup_compute_volume(context,\n1267:            self.volume_manager.remove_compute_volume(context,\n1289:        self.volume_manager.remove_compute_volume(context, volume_id)\n1306:        self.volume_manager.remove_compute_volume(context, volume_id)\n1401:                self.volume_manager.setup_compute_volume(context, v['id'])\n1511:                self.volume_manager.remove_compute_volume(ctxt, vol['id'])", 
            "date_created": "2011-10-11 19:07:40.379268+00:00", 
            "author": "https://api.launchpad.net/1.0/~nati-ueno"
        }, 
        {
            "content": "I think this just got merged in by Vishy in one of his volume cleanup branches", 
            "date_created": "2011-10-12 19:25:31.161037+00:00", 
            "author": "https://api.launchpad.net/1.0/~rnirmal"
        }, 
        {
            "content": ">Nirmal Ranganathan\nThanks\nI confirmed it is merged to Essex branch.\nBut I think it is also needed for Diablo Stable Branch.\n\nhttps://github.com/openstack/nova/blob/2011.3/nova/compute/manager.py\n", 
            "date_created": "2011-10-12 20:09:10+00:00", 
            "author": "https://api.launchpad.net/1.0/~nati-ueno"
        }, 
        {
            "content": "Ok cool.. yeah not sure what all was going to get backported to Diablo.\n\nOn Wed, Oct 12, 2011 at 3:09 PM, Nachi Ueno <email address hidden>wrote:\n\n> >Nirmal Ranganathan\n> Thanks\n> I confirmed it is merged to Essex branch.\n> But I think it is also needed for Diablo Stable Branch.\n>\n> https://github.com/openstack/nova/blob/2011.3/nova/compute/manager.py\n>\n> --\n> You received this bug notification because you are subscribed to the bug\n> report.\n> https://bugs.launchpad.net/bugs/872445\n>\n> Title:\n>  nova.compute.manager calls methods in nova.volume.manager directly\n>\n> Status in OpenStack Compute (Nova):\n>  New\n> Status in OpenStack QA:\n>  New\n>\n> Bug description:\n>  nova.compute.manager calls methods in nova.volume.manager directly.\n>  This must be via api module.\n>\n>  146:        self.volume_manager =\n> utils.import_object(FLAGS.volume_manager)\n>  1230:        dev_path = self.volume_manager.setup_compute_volume(context,\n> volume_id)\n>  1241:        dev_path = self.volume_manager.setup_compute_volume(context,\n>  1267:            self.volume_manager.remove_compute_volume(context,\n>  1289:        self.volume_manager.remove_compute_volume(context, volume_id)\n>  1306:        self.volume_manager.remove_compute_volume(context, volume_id)\n>  1401:                self.volume_manager.setup_compute_volume(context,\n> v['id'])\n>  1511:                self.volume_manager.remove_compute_volume(ctxt,\n> vol['id'])\n>\n> To manage notifications about this bug go to:\n> https://bugs.launchpad.net/nova/+bug/872445/+subscriptions\n>\n", 
            "date_created": "2011-10-12 20:31:35+00:00", 
            "author": "https://api.launchpad.net/1.0/~rnirmal"
        }
    ], 
    "closed": "2011-11-17 10:02:45.328905+00:00"
}